### PR TITLE
feat: Add GPG_SECRET_KEY and GPG_PASSPHRASE secrets for craft

### PR DESCRIPTION
**Pull request by kamilogorek**
This PR is a prerequisite for https://github.com/getsentry/craft/pull/346

To generate a valid secret key:

```terminal
gpg --quick-generate-key bot@sentry.io
gpg --armor --export-secret-keys bot@sentry.io | pbcopy
```

Paste copied key as `GPG_SECRET_KEY` and entered password as `GPG_PASSPHRASE`.

### CODE REVIEW

**Review by marandaneto — APPROVED** [review]
Is the key already available in the docker image?

Thanks @kamilogorek

**Comment by kamilogorek** [comment]
No, @bruno-garcia will generate it and add once he is awake.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -91,6 +91,8 @@ jobs:
           AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           NUGET_API_TOKEN: ${{ secrets.NUGET_API_TOKEN }}
+          GPG_SECRET_KEY: ${{ secrets.GPG_SECRET_KEY }}
+          GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
```
Comment (chadwhitacre) [inline .github/workflows/publish.yml:95]: I believe the presenting case here is the Maven target (https://github.com/getsentry/craft/pull/346). Will we ever want GPG signing for other targets? Does it make sense to use a global GPG key for all targets as the var name here suggests? Do we want to scope this down to Maven or somesuch?

Comment (kamilogorek) [reply .github/workflows/publish.yml:95]: I was thinking about this, but I don't see a reason not to use a generic key here. Bruno mentioned something about .net, there is signing for iOS as well, so we may need it in the future.

Comment (bruno-garcia) [reply .github/workflows/publish.yml:95]: .NET code signing won't be done with GPG (it's a code signing cert and the dotnet built-in tooling)

Comment (chadwhitacre) [reply .github/workflows/publish.yml:95]: Okay, sounds like we're on it.

**Review by chadwhitacre — APPROVED** [review]

**Pull request merged**